packages/babel-generator/src/printer.ts

```diff
@@ -799,22 +798,17 @@ class Printer {
           this.format.retainLines ? 0 : this._buf.getCurrentColumn(),
         );
         val = val.replace(/\n(?!$)/g, `\n${" ".repeat(indentSize)}`);
-
-        maybeNewline = true;
       }
     } else if (!this._noLineTerminator) {
       val = `//${comment.value}\n`;
-      maybeNewline = true;
+    } else {
+      val = `/*${comment.value}*/`;
```

Comment:
This case is for when we had a line comment but we cannot have a line terminator? When can it happen?

Reply:
It's in the associated issue. I also added to the test.
You said that and it reminded me that it doesn't contain newlines!😄

Reply:
Testing doesn't seem to reflect this.😕

Tested on my local.
```ts
const test = (): {
  /* test*/foo: string;
} => {
  return {
    foo: "bar"
  };
};

export default test;
```

Reply:
https://github.com/babel/babel/pull/14758
https://github.com/babel/babel/pull/14762

Oh no, after a long time, I finally found the reason.
Only the first pr was published, and the second was not.

So this test is useless, we need to find another one.

By the way, can you release a new version of `jest-light-runner`?
Filtering of tests can be helpful in debugging generators and parsers.